resource/graph_node.go

```diff
@@ -510,6 +539,15 @@ func (w *GraphNode) ResourceStatus() Status {
 	return w.resourceStatus()
 }
 
+func (w *GraphNode) getLoggerOrGlobal() logging.Logger {
+	if w.logger == nil {
+		// This node has not yet been configured with a logger - use the global logger as
+		// a fall-back.
```

Comment:
[q] Any idea when this happens? Just want to get a sense of when resources are logging in the absence of a logger. Seems like a violation of my mental model of graph node loggers...

Reply:
i forgot the specifics but there is a lag between when a node is first created and when it receives an initial logger. if we remove the `w.logger == nil` guards from this file then we will get a nil-access panic pretty quickly. 

I suppose we should use this function through the file instead of relying on the guard to get more uniform behavior -- happy to do this as a follow-up.